pkg/prometheus/rules.go

```diff
@@ -0,0 +1,54 @@
+package prometheus
+
+import (
+	monv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	monitoringclient "github.com/coreos/prometheus-operator/pkg/client/versioned"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var ruleLog = logf.Log.WithName("common_prometheusRule")
+
+func newMonitoringClient() (*monitoringclient.Clientset, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", "")
+	if err != nil {
+		return nil, err
+	}
+	monitoringClient, err := monitoringclient.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return monitoringClient, err
+}
+
+// CreateOrUpdatePrometheusRule Creates or Updates prometheusRule object
+func CreateOrUpdatePrometheusRule(p *monv1.PrometheusRule) error {
+	mclient, err := newMonitoringClient()
+	pclient := mclient.MonitoringV1().PrometheusRules(p.GetNamespace())
+	oldRule, err := pclient.Get(p.GetName(), metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		_, err := pclient.Create(p)
+		if err != nil {
+			ruleLog.Error(err, "creating PrometheusRule object failed", "Prometheus Namespace: ", p.ObjectMeta.Namespace)
```

Comment:
create a function which will return you a message. I can see the message is same for creating, retrieving and updating.

Reply:
Why add overhead of one more function?
Messages needs to be worked on, but are of very low priority for now.

Reply:
it's actually not a overhead. Its good to have a generic function so we don't need to change at multiple places the same things. Also reduces the chances of errors.

Reply:
ack